node/src/actors/chain_manager/mining.rs

```diff
@@ -108,6 +108,19 @@ impl ChainManager {
             .consensus_constants
             .epochs_with_initial_difficulty;
 
+        let (total_identities, replication_factor) =
+            if after_second_hard_fork(current_epoch, get_environment()) {
+                (
+                    u32::try_from(rep_engine.trs().num_identities()).unwrap(),
```

Comment:
Are we sure that `rep_eng.trs().num_identities()` will never be 0? Maybe we should add a minimum difficulty to prevent saturating the network with candidates when this value is low.